tests/sles4sap/trento/test_trento_deploy.pm

```diff
@@ -8,33 +8,49 @@ use Mojo::Base 'publiccloud::basetest';
 use base 'consoletest';
 use strict;
 use testapi;
-use mmapi 'get_current_job_id';
+use utils qw(script_retry);
+use base 'trento';
 
-use constant TRENTO_AZ_PREFIX => 'openqa-trento';
 
 sub run {
     my ($self) = @_;
     die "Only AZURE deployment supported for the moment" unless check_var('PUBLIC_CLOUD_PROVIDER', 'AZURE');
     $self->select_serial_terminal;
-    my $job_id = get_current_job_id();
 
-    my $resource_group = TRENTO_AZ_PREFIX . "-rg-$job_id";
-    my $machine_name = TRENTO_AZ_PREFIX . "-vm-$job_id";
+    my $resource_group = $self->get_resource_group;
 
     # check if VM is still there :-)
     assert_script_run("az vm list -g $resource_group --query \"[].name\"  -o tsv", 180);
 
-    my $machine_ip = script_output("az vm show -d -g $resource_group -n $machine_name --query \"publicIps\" -o tsv", 180);
+    # get deployed version from the cluster
+    my $machine_ip = $self->az_get_vm_ip;
+    my $kubectl_pods = script_output($self->az_vm_ssh_cmd('kubectl get pods', $machine_ip), 180);
+    foreach my $row (split(/\n/, $kubectl_pods)) {
+        if ($row =~ m/trento-server-web/) {
+            my $pod_name = (split /\s/, $row)[0];
+            my $trento_ver_cmd = $self->az_vm_ssh_cmd("kubectl exec --stdin $pod_name -- /app/bin/trento version", $machine_ip);
+            script_run($trento_ver_cmd, 180);
+        }
+    }
 
     # test if the web page is reachable on http
-    assert_script_run("curl -k  http://" . $machine_ip . "/");
+    my $trento_url = 'http://' . $machine_ip . '/';
```

Comment:
Also a general suggestion:
```suggestion
    my $trento_url = "http://$machine_ip/";
```
I personally find interpolation more readable in many cases, where it's possible.
For longer concatenations I can also recommend to use [sprintf](https://perldoc.perl.org/functions/sprintf)